safekeeper/src/wal_storage.rs

```diff
@@ -504,69 +509,125 @@ pub struct WalReader {
     timeline_dir: PathBuf,
     wal_seg_size: usize,
     pos: Lsn,
-    file: Option<File>,
+    wal_segment: Option<Pin<Box<dyn AsyncRead>>>,
+
+    enable_remote_read: bool,
+    // S3 will be used to read WAL is LSN it not available locally
```

Comment:
looks like a typo: `is LSN it not available locally` -> `if LSN is not available locally`

Reply:
I wonder, if code suggestions work

```suggestion
    // S3 will be used to read WAL if LSN is not available locally
```

Reply:
Yes, it works. Thanks!